pystiche_papers/sanakoyeu_et_al_2018/_data.py

```diff
@@ -0,0 +1,216 @@
+    BASE_URL = "https://hcicloud.iwr.uni-heidelberg.de/index.php/s/NcJj2oLBTYuT1tf/download?path=%2F&files="
+    MD5_CHECKSUMS = {
+        "berthe-morisot": "0f61db2107b16a86fe38caff1a2c4125",
+        "edvard-munch": "8a75855098b412f574540cb01a7e2dda",
+        "el-greco": "fb9ad7ee563d5c6ed3ea9498567e655c",
+        "ernst-ludwig-kirchner": "35137d813ee58505266e46e24813dc42",
+        "jackson-pollock": "6fbece7f00da43b9492b835fac4cc7f8",
+        "monet_water-lilies-1914": "bba66d8a3fff62a4b0b714b215636d01",
+        "nicholas-roerich": "1683e2b1e4af7853bdfe1b63f8db0e26",
+        "pablo-picasso": "54b99819e79f522fde04ba324215814a",
+        "paul-cezanne": "7c1d841f98e21f5de3ba69a3aa25b3bd",
+        "sample_photographs": "173967574bf236a21044ef593286dd6f",
+        "samuel-peploe": "de82d2654ee437abdd8a6b3cdf458bfd",
+        "vincent-van-gogh_road-with-cypresses-1890": "6e629d6a03c65bc510bba8b619aad291",
+        "wassily-kandinsky": "b506040ee2d038b8f3767125d04bde5f",
+    }
+    STYLES = tuple(sorted(MD5_CHECKSUMS.keys()))
+
+    def __init__(
+        self,
+        root: str,
+        style: str,
+        transform: Optional[nn.Module] = None,
+        download: bool = False,
+    ) -> None:
+        self.root = root = path.abspath(path.expanduser(root))
+        self.style = self._verify_style(style)
+
+        if download:
+            self.download()
+
+        super().__init__(self.sub_dir, transform=transform)
+        self.root = root
+
+    def _verify_style(self, style: str) -> str:
+        return verify_str_arg(style, "style", self.STYLES)
+
+    @property
+    def sub_dir(self) -> str:
+        return path.join(self.root, self.style)
+
+    @property
+    def archive(self) -> str:
+        return f"{self.sub_dir}.tar.gz"
+
+    @property
+    def md5(self) -> str:
+        return self.MD5_CHECKSUMS[self.style]
+
+    @property
+    def url(self) -> str:
+        return f"{self.BASE_URL}{path.basename(self.archive)}"
+
+    def download(self) -> None:
+        if path.exists(self.sub_dir):
+            msg = (
+                f"The directory {self.sub_dir} already exists. If you want to "
+                "re-download the images, delete the folder."
+            )
+            raise RuntimeError(msg)
+
+        download_and_extract_archive(
+            self.url, self.root, filename=self.archive, md5=self.md5
+        )
+
+
+def style_dataset(
+    root: str,
+    style: str,
+    transform: Optional[nn.Module] = None,
+    download: bool = False,
+) -> WikiArt:
+    if transform is None:
+        transform = image_transform()
+    return WikiArt(root, style, transform=transform, download=download)
+
+
+# TODO: replace this with torchvision.datasets.Places365 as soon as
+#  https://github.com/pytorch/vision/pull/2610 is part of a release
+def content_dataset(
+    root: str, transform: Optional[transforms.Transform] = None,
+) -> ImageFolderDataset:
+    if transform is None:
+        transform = image_transform()
```

Comment:
I think I have mentioned this before, but the images of the Places365 are enlarged to twice the size again before this `image_transform`. See [here](https://github.com/CompVis/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/prepare_dataset.py#L133).

Reply:
Only for these images or for all images (including style) in general?

Reply:
This is only for the dataset Places365, but this is not mentioned in the paper.